packages/platform-browser/src/browser/browser_adapter.ts

```diff
@@ -89,11 +89,9 @@ function getBaseElementHref(): string|null {
   return baseElement ? baseElement.getAttribute('href') : null;
 }
 
-// based on urlUtils.js in AngularJS 1
-let urlParsingNode: HTMLAnchorElement|undefined;
-function relativePath(url: any): string {
-  urlParsingNode = urlParsingNode || document.createElement('a');
-  urlParsingNode.setAttribute('href', url);
-  const pathName = urlParsingNode.pathname;
+function relativePath(url: string): string {
+  // The base URL doesn't really matter, we just need it so relative paths have something
+  // to resolve against. In the browser `HTMLBaseElement.href` is always absolute.
+  const pathName = new URL(url, 'http://a').pathname;
   return pathName.charAt(0) === '/' ? pathName : `/${pathName}`;
```

Comment:
`new URL().pathname` always returns with a leading forward slash.

See: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname 

```suggestion
  return new URL(url, 'http://a').pathname;
```